dali/operators/geometry/affine_transforms/transform_shear.cc

```diff
@@ -163,16 +174,16 @@ class TransformShearCPU
   }
 
  private:
-  int InferNumDims(const Argument<std::vector<float>> &arg) {
-    DALI_ENFORCE(arg[0].size() == 2 || arg[0].size() == 6,
+  int InferNumDims(const ArgValue<float, 1> &arg) {
+    DALI_ENFORCE(arg[0].num_elements() == 2 || arg[0].num_elements() == 6,
       make_string("Unexpected number of elements in ``", arg.name(), "`` argument. "
-                  "Expected 2 or 6 arguments. Got: ", arg[0].size()));
-    return arg[0].size() == 6 ? 3 : 2;
+                  "Expected 2 or 6 arguments. Got: ", arg[0].num_elements()));
+    return arg[0].num_elements() == 6 ? 3 : 2;
   }
 
-  Argument<std::vector<float>> shear_;
-  Argument<std::vector<float>> angles_;
-  Argument<std::vector<float>> center_;
+  ArgValue<float, 1> shear_;
```

Comment:
```suggestion
  ArgValue<float, DynamicDimensions> shear_;
```
I think shear can be a matrix, can't it?